### ISSUE TITLE
Updating a device in core-metadata requires much detail

### ISSUE THREAD

**Issue by iain-anderson** [edgexfoundry/edgex-go#1505]
# 🐞 Bug Report

### Affected Services
The issue is located in: core-metadata

### Description and Minimal Reproduction

Run a device sdk example and stop it immediately, in order to create a device
Attempt to update the device as suggested in sample operation given in core-metadata.raml:
```
curl -X PUT -d '{"id":"91e23e0d-e1d2-4a02-ad3b-3d3549a17414", "description":"living room thermostat","adminState":"locked"}'  0:48081/api/v1/device
```

## 🔥 Exception or Error
<pre><code>
no supporting protocol specified for device
</code></pre>

In the core-metadata log:

<pre><code>
level=ERROR ts=2019-07-05T14:13:59.849392408Z app=edgex-core-metadata source=rest_device.go:115 msg="no supporting protocol specified for device"
</code></pre>

## 🌍  Your Environment
**Deployment Environment:**
Ubuntu 1804, x86_64, not containerized.

**EdgeX Version:**
1.0 and 1.1

**Anything else relevant?**

The update can be performed if the protocols, adminState, operatingState, service (name only) and profile (name only) fields are given:
```
curl -X PUT -d '{"id":"b473304b-5581-40f1-a9020bb350c69", "protocols":{"Other":{"Address":"someSensorEndpoint"}},"adminState":"UNLOCKED","operatingState":"ENABLED","service":{"name":"device-template"},"profile":{"name":"TemplateSensor"}}' 0:48081/api/v1/device
```

I assume that we are attempting to validate the PUT payload as a complete Device definition rather than a subset of fields that are to be updated.

**Comment by tsconn23** [comment]
@iain-anderson This is a known issue we've talked about for some time in the Core WG. The problem is we can't really change this until we go to a version 2.0.0 API.

Related issues to this are
https://github.com/edgexfoundry/go-mod-core-contracts/issues/40
https://github.com/edgexfoundry/go-mod-core-contracts/issues/27

And also the project backlog found here
https://github.com/edgexfoundry/edgex-go/projects/10

My proposal, which I've discussed many times during the Core WG, is that we refactor toward request-specific types rather than passing these large state models which have their own integrity constraints that are frequently meaningless when used as request/response models rather than state representations.

Therefore, I suggest this be made into a feature, not a bug, and put onto the Geneva roadmap.
